opaque_keys/edx/locator.py

```diff
@@ -1620,3 +1621,59 @@ def html_id(self) -> str:
         # HTML5 allows ID values to contain any characters at all other than spaces.
         # These key types don't allow spaces either, so no transform is needed.
         return str(self)
+
+
+class LibCollectionLocator(CheckFieldMixin, LibCollectionKey):
+    """
+    When serialized, these keys look like:
+        lib-collection:org:lib:collection-id
+    """
+    CANONICAL_NAMESPACE = 'lib-collection'
+    KEY_FIELDS = ('org', 'lib', 'usage_id')
+    org: str
+    lib: str
+    lib_key: LibraryLocatorV2
+    usage_id: str
+
+    __slots__ = KEY_FIELDS
+    CHECKED_INIT = False
+
+    # Allow usage IDs to contian unicode characters
+    USAGE_ID_REGEXP = re.compile(r'^[\w\-.]+$', flags=re.UNICODE)
```

Comment:
This is the collection id (a number), but I'm unsure if we want to make it stricter here.

Maybe for clarity? What do you think?

Reply:
Wait, it's the ID? As in the primary key? We can't use that as part of the key if there's ever an intention to make it importable into a different instance.

Reply:
At the moment we don't have any other identifier for collections; we'll need to add either a UUID or a slug (or both). If a slug, it only needs to be unique within a library. I prefer slugs for readability. For now the slug can be auto-generated by [slugifying](https://docs.djangoproject.com/en/5.1/ref/utils/#django.utils.text.slugify) the collection name and appending a number until it's unique.

Reply:
Yeah, I think a slug is fine, as long as we write it to the database–say, a `key` field on `Collection` with a unique constraint on the combo of that and the LearningPackage (like how PublishableEntities work today).

Reply:
Cool, let's do that ^ @rpenido . Sorry for not realizing the need for a slug sooner. CC @pomegranited